docs/StardustDocs/topics/schemas/schemas.md

```diff
@@ -0,0 +1,140 @@
+
+For example, consider a simple hierarchical DataFrame from
+<resource src="example.csv"></resource>.
+
+This DataFrame consists of two columns:
+- `name`, which is a `String` column
+- `info`, which is a [column group](DataColumn.md#columngroup) containing two nested [value columns](DataColumn.md#valuecolumn):
+    - `age` of type `Int`
+    - `height` of type `Double`
+
+<table>
+  <thead>
+    <tr>
+      <th>name</th>
+      <th colspan="2">info</th>
+    </tr>
+    <tr>
+      <th></th>
+      <th>age</th>
+      <th>height</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Alice</td>
+      <td>23</td>
+      <td>175.5</td>
+    </tr>
+    <tr>
+      <td>Bob</td>
+      <td>27</td>
+      <td>160.2</td>
+    </tr>
+  </tbody>
+</table>
+
+The data schema corresponding to this DataFrame can be represented as:
+
+```kotlin
+// Data schema of the "info" column group
+@DataSchema
+data class Info(
+    val age: Int,
+    val height: Float
+)
+
+// Data schema of the entire DataFrame
+@DataSchema
+data class Person(
+    val info: Info,
+    val name: String
+)
+```
+
+[Extension properties](extensionPropertiesApi.md) for `DataFrame<Person>`  
+are generated based on this schema and allow accessing columns
+or using them in operations:
+
+```kotlin
+// Assuming `df` has type `DataFrame<Person>`
+
+// Get "age" column from "info" group
+df.info.age
+
+// Select "name" and "height" columns
+df.select { name and info.height }
+
+// Filter rows by "age"
+df.filter { age >= 18 }
+```
+
+See [](extensionPropertiesApi.md) for more information.
+
+
+## Schemas retrieving
+
+Defining a data schema manually can be difficult, especially for dataframes with many columns or deeply nested structures, and may lead to mistakes in column names or types. Kotlin DataFrame provides several methods for generating data schemas.
+
+* [**`generate..()` methods**](DataSchemaGenerationMethods.md) are extensions for [`DataFrame`](DataFrame.md) that generate a code string representing its `DataSchema`.
+
+* [**Kotlin DataFrame Compiler Plugin**](Compiler-Plugin.md) **cannot automatically infer** a 
+data schema from external sources such as files or URLs.
+However, it **can** infer the schema if you construct the [`DataFrame`](DataFrame.md) 
+manually — that is, by explicitly declaring the columns using the API.
+It will also **automatically update** the schema during operations that modify the structure of the DataFrame.
+
+> For best results when working with the Compiler Plugin, it's recommended to 
+> generate the initial schema using one of 
+> the [`generate..()` methods](DataSchemaGenerationMethods.md).
+> Once generated, the Compiler Plugin will automatically keep the schema up to date 
+> after any operations that change the structure of the DataFrame.
+
+### Plugins
+
+> The current Gradle plugin is **under consideration for deprecation** and may be officially marked as deprecated in future releases.  
+> The KSP plugin is currently non-functional.
```

Comment:
no it isn't. It should still work, but, it's also gonna be deprecated, otherwise we'd need to migrate to KSP2